assets/helpers/tracking/googleTagManager.js

```diff
@@ -32,25 +35,24 @@ function getContributionValue() {
   return storage.getSession('contributionValue') || 0;
 }
 
+function pushBasicDimensions() {
+  window.googleTagManagerDataLayer = [{
+    // orderId anonymously identifies this user in this session.
+    // We need this to prevent page refreshes on conversion pages being
+    // treated as new conversions
+    orderId: getDataValue('orderId', uuidv4),
+    currency: getDataValue('currency', getCurrency),
+    value: getContributionValue(),
+  }];
+}
+
+// ----- Exports ---//
 
-// ----- Run ----- //
-
-window.googleTagManagerDataLayer = [{
-  // orderId anonymously identifies this user in this session.
-  // We need this to prevent page refreshes on conversion pages being
-  // treated as new conversions
-  orderId: getDataValue('orderId', uuidv4),
-  currency: getDataValue('currency', getCurrency),
-  value: getContributionValue(),
-}];
-
-// Google Tag Manager
-/* eslint-disable */
-(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
-  new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
-  j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
-// $FlowFixMe
-  'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
-})(window,document,'script','googleTagManagerDataLayer','GTM-5PKPPQZ');
-/* eslint-enable */
-// End Google Tag Manager
+export function pushDimensions(dimensions: Dimensions) {
+  const dataLayer = window.googleTagManagerDataLayer;
+  dataLayer.push(dimensions);
+}
+
+export function init() {
+  pushBasicDimensions();
```

Comment:
Would I be correct to think that this doesn't actually `push` anything, but loads the data into the page, ready for the call by the GTM script to actually push it? If so, maybe a name like `setBasicDimensions` would be more accurate.

Reply:
Yeap, I updated it! Thanks Justin!